daemon/src/main.rs

```diff
@@ -739,12 +822,13 @@ fn is_daemon_running() -> Result<bool, String> {
             if std::process::id() == pid {
                 continue;
             }
+
             let mut entry_path = entry.path();
             entry_path.push("cmdline");
             if let Ok(cmd) = std::fs::read_to_string(entry_path) {
                 let mut args = cmd.split(&[' ', '\0']);
                 if let Some(arg0) = args.next() {
-                    if arg0.ends_with("swww-daemon") {
+                    if arg0.ends_with("swww-daemon") && check_fds(fd_num, entry.path())? {
                         return Ok(true);
                     }
                 }
```

Comment:
If you implement what I've commented above, this proc traversal code will become redundant, and you can delete it.